CONTRIBUTING.md

```diff
@@ -0,0 +1,227 @@
+
+<details>
+
+### Steps to reproduce
+
+1. On Ubuntu 22.04
+2. Run `subspace-farmer-ubuntu-x86_64-gemini-1b-2022-jun-18 farm` from official release
+3. Observe an error
+
+### Expected result
+
+Farmer starts successfully
+
+### What happens instead
+
+This error
+```
+error: The following required arguments were not provided:
+    --reward-address <REWARD_ADDRESS>
+    --plot-size <PLOT_SIZE>
+
+USAGE:
+    subspace-farmer-ubuntu-x86_64-gemini-1b-2022-jun-18 [OPTIONS] --reward-address <REWARD_ADDRESS> --plot-size <PLOT_SIZE>
+
+For more information try --help
+```
+
+</details>
+
+That is an invalid bug report, of course, but you get the idea of what it should look like.
+
+Try to provide as much useful information as possible, but not too much and without rambling, so that developers can
+quickly diagnose and address it.
+
+### Improvements and feature requests
+
+When there is something that can be improved or a feature is lacking, it is probably a good idea to start with the forum
+and search for similar ideas and if there isn't a topic already, create one to see if this would be a welcome change.
+
+Again, try to discuss with wider audience whether it is a good idea before reaching out to maintainers directly.
+
+When it is necessary to create an issue here, try to describe in a concise way what would the improvement or new feature
+look like, potentially using a similar template as for [reporting bugs](#bugs) describing expected user flow there.
+
+## Code contributions
+
+Code contributions are more involved, but the simplest rule here is to observe what maintainers do, how existing code
+looks like and mimic that, but we provide more specific requirements below.
+
+### Code style
+
+We follow traditional Rust code style with some clarifications and set in `rustfmt.toml`. We also follow all Clippy
+lints (with rare suppressions). As such every commit should successfully pass `cargo fmt` checks and produce zero
+warnings when running `cargo clippy`. This rule includes not just library and application code, but also examples and
+tests.
+
+Every file should end with one and exactly one new line, configure your IDE to insert it automatically. Lines should
+not end with whitespaces, this includes code, comments, documentation (`*.md` files), configs and everything else.
+
+Pay attention to existing code and mimic its structure, for instance we keep dependencies sorted order in `Cargo.toml`
+and if you submit changes where new dependency is added out-of-order you're introducing unnecessary entropy and irritate
+maintainers.
+
+If there is an option between being smart with a tricky piece of code or obvious, prefer obvious. Write idiomatic Rust
+as good as you can, this often allows to avoid needing `.unwrap()`.
+
+Struct definition and its implementation should be in the same file, ideally with nothing else in between.
+
+In blockchain code use of `.unwrap()` is strictly forbidden and must never be used (except test code), use `.expect()`
```

Comment:
Should we just enable this lint?
https://rust-lang.github.io/rust-clippy/master/index.html#unwrap_used

Not as a part of this pr of course

Reply:
Absolutely! It is much easier to enforce if it doesn't compile unless follow the rules :slightly_smiling_face: 
But allow in tests, there is an option for that there. Feel free to send a PR, but this is not a high priority.